silkworm/core/protocol/clique_rule_set.cpp

```diff
@@ -30,11 +30,11 @@ static evmc::address ecrecover(const BlockHeader& header) {
     evmc::address beneficiary = evmc::address{};
 
     evmc::bytes32 seal_hash = header.hash(false, true);
-    Bytes signature = header.extra_data.substr(header.extra_data.length() - kExtraSealSize, kExtraSealSize - 1);
-    bool odd_y_parity = header.extra_data[header.extra_data.length() - 1] != 0;
+    ByteView signature{&header.extra_data[header.extra_data.length() - kExtraSealSize], kExtraSealSize - 1};
+    uint8_t recovery_id{header.extra_data[header.extra_data.length() - 1]};
 
     static secp256k1_context* context{secp256k1_context_create(SILKWORM_SECP256K1_CONTEXT_FLAGS)};
-    if (!silkworm_recover_address(beneficiary.bytes, seal_hash.bytes, signature.c_str(), odd_y_parity, context)) {
+    if (!silkworm_recover_address(beneficiary.bytes, seal_hash.bytes, signature.data(), recovery_id, context)) {
```

Comment:
Currently, the `silkworm_recover_address` function has `bool odd_y_parity` as fourth formal parameter. It calls the `recover` function, which in turns calls `secp256k1_ecdsa_recoverable_signature_parse_compact` passing the received `odd_y_parity` as actual parameter for `int recid`.

So, I see why you are changing the parameter type and name here. I think in another PR we should change also the formal parameter type and name in `silkworm_recover_address` trying to accommodate all needs (i.e. somewhere else it's called with `v != 27`).

Reply:
In this PR I've changed the parameter of `silkworm_recover_address` to `uint8_t recovery_id`. It's called with `v != 27` in the ecrec_run precompile, where `v` is checked to be either 27 or 28.

Reply:
Sorry, I completely missed the change is already in this PR.